app/registries/foreman/plugin.rb

```diff
@@ -346,12 +355,13 @@ def add_all_permissions_to_default_roles
 
     def pending_migrations
       return true if Foreman.in_setup_db_rake?
+      return @pending_migrations unless @pending_migrations.nil?
 
-      pending_migrations = ActiveRecord::Base.connection.migration_context.needs_migration?
+      @pending_migrations = ActiveRecord::MigrationContext.new(migrations_paths, ActiveRecord::SchemaMigration).needs_migration?
```

Comment:
It looks like this caused https://projects.theforeman.org/issues/30821. If a plugin has no migrations but uses permissions, this fails in packaging. I think it should still take the main DB into account, as well as plugins.